### PR TITLE
Link Control: Fix scroll bar on long links

### LINKED ISSUE #36961: Long URLs cause horizontal scroll bar in the Link UI
### Description

If you add a very long URL via the Link UI then it causes a horiztonal scroll bar to appear.

### Step-by-step reproduction instructions

1. New Post
2. Add link to a very long URL `https://wordpress.slack.com/archives/C02QB2JS7/p1637762376489200`.
3. Click on Link to show preview.
4. See scrollbar.

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/444434/143880184-932c7d40-8a17-42c2-af43-31c963bb3cc0.mp4

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by walbo**
## Description
If you add a very long URL via the Link UI then it causes a horiztonal scroll bar to appear. Make sure the URL is truncated.

Fixes #36961

## How has this been tested?
1. Added a new Post
1. Added link to a very long URL https://wordpress.slack.com/archives/C02QB2JS7/p1637762376489200.
1. Clicked on Link to show preview.
1. Confirmed that there is no scroll bar.

## Screenshots 
image

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-editor/src/components/link-control/style.scss [hunk 1]
```diff
@@ -192,6 +192,7 @@ $preview-image-height: 140px;
 		flex-direction: row;
 		align-items: flex-start;
 		margin-right: $grid-unit-10;
+		min-width: 0;
```
Comment (getdave) [inline packages/block-editor/src/components/link-control/style.scss:195]: I think we should include a comment to explain what this does and why it's necessary. That would help avoid regressions.

File: packages/block-editor/src/components/link-control/style.scss [hunk 2]
```diff
@@ -224,6 +225,10 @@ $preview-image-height: 140px;
 		}
 	}
 
+	.block-editor-link-control__search-item-details {
+		min-width: 0;
+	}
+
```
Comment (getdave) [inline packages/block-editor/src/components/link-control/style.scss:231]: As above, I think we should include a comment to explain what this does and why it's necessary. That would help avoid regressions.

**Review by getdave — COMMENTED** [review]
Thanks for PR. Very much appreciated 🙇 


 I have this on my list to test.

Left some comments (about comments!) in the meantime.

File: packages/block-editor/src/components/link-control/style.scss (hunk 1 above)
Comment (walbo) [reply packages/block-editor/src/components/link-control/style.scss:195]: 👍 

I can add some comments later today (unless the alternative PR #36993 is merged).

**Comment by getdave** [comment]
As this PR truncates some of the strings I'd like to make everyone aware of this comment

https://github.com/WordPress/gutenberg/pull/36993#issuecomment-982665722

**Comment by walbo** [comment]
Closing this in favor of #36993

**Comment by getdave** [comment]
Thank you for your work here @walbo - very much appreciated.

**Pull request closed without merging**